### PR TITLE
Closes #111  | Add Thai Constitution

### LINKED ISSUE #111: Create dataset loader for Thai Constitution Dataset
Dataloader name: `thai_constitution/thai_constitution.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?thai_constitution

|     Dataset| thai_constitution |
|-------------|---|
| Description | A collection of snippets from the Thai constitution from 1932 and retrieved from the Thai Office of the Council of State (http://www.krisdika.go.th/). |
| Subsets | - |
| Languages | tha |
| Tasks | Language Modeling, Language Identification |
| License | Apache license 2.0 (apache-2.0) |
| Homepage | https://github.com/PyThaiNLP/Thai-constitution-corpus/tree/master/data |
| HF URL | - |
| Paper URL | https://github.com/PyThaiNLP/Thai-constitution-corpus/tree/master/data |

### PR DESCRIPTION
**Pull request by ilhamfp**
Closes #111 

Test passed:
image

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/my_dataset/my_dataset.py` (please use only lowercase and underscore for dataset naming).
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [x] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [x] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

### CODE REVIEW

**Review by holylovenia — CHANGES_REQUESTED** [review]
Thanks for the dataloader, @ilhamfp! It's good as usual. 👍 One nit, could you please change `_SUPPORTED_LANGS` to `_LANGUAGES`?

File: seacrowd/sea_datasets/thai_constitution/thai_constitution.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+        filepaths = [Path(dl_manager.download(url)) for url in _URLS]
+
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepaths": filepaths},
+            ),
+        ]
+
+    def _generate_examples(self, filepaths: list[Path]) -> Tuple[int, Dict]:
```
Comment (yongzx) [inline seacrowd/sea_datasets/thai_constitution/thai_constitution.py:118]: it should be `List` instead of `list`.

**Review by yongzx — CHANGES_REQUESTED** [review]

File: seacrowd/sea_datasets/thai_constitution/thai_constitution.py (hunk 1 above)
Comment (ilhamfp) [reply seacrowd/sea_datasets/thai_constitution/thai_constitution.py]: Good catch! Updated

**Comment by yongzx** [comment]
Thanks for the quick fix! One last thing, can you replace `ssp` with `SEACROWD_SCHEMA_NAME` instead? 

See https://github.com/SEACrowd/seacrowd-datahub/pull/153#discussion_r1445956869

File: seacrowd/sea_datasets/thai_constitution/thai_constitution.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name="thai_constitution_source",
+            version=datasets.Version(_SOURCE_VERSION),
+            description="Thai constitution source schema",
+            schema="source",
+            subset_id="thai_constitution",
+        ),
+        SEACrowdConfig(
+            name="thai_constitution_seacrowd_ssp",
```
Comment (yongzx) [inline seacrowd/sea_datasets/thai_constitution/thai_constitution.py:79]: Would be great to replace `ssp` with `self.SEACROWD_SCHEMA_NAME`

File: seacrowd/sea_datasets/thai_constitution/thai_constitution.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+    "https://raw.githubusercontent.com/PyThaiNLP/Thai-constitution-corpus/master/data/%E0%B8%A3%E0%B8%B1%E0%B8%90%E0%B8%98%E0%B8%A3%E0%B8%A3%E0%B8%A1%E0%B8%99%E0%B8%B9%E0%B8%8D%E0%B9%81%E0%B8%AB%E0%B9%88%E0%B8%87%E0%B8%A3%E0%B8%B2%E0%B8%8A%E0%B8%AD%E0%B8%B2%E0%B8%93%E0%B8%B2%E0%B8%88%E0%B8%B1%E0%B8%81%E0%B8%A3%E0%B9%84%E0%B8%97%E0%B8%A2%20%E0%B8%9E%E0%B8%B8%E0%B8%97%E0%B8%98%E0%B8%A8%E0%B8%B1%E0%B8%81%E0%B8%A3%E0%B8%B2%E0%B8%8A%202475%20%E0%B9%81%E0%B8%81%E0%B9%89%E0%B9%84%E0%B8%82%E0%B9%80%E0%B8%9E%E0%B8%B4%E0%B9%88%E0%B8%A1%E0%B9%80%E0%B8%95%E0%B8%B4%E0%B8%A1%20%E0%B8%9E%E0%B8%B8%E0%B8%97%E0%B8%98%E0%B8%A8%E0%B8%B1%E0%B8%81%E0%B8%A3%E0%B8%B2%E0%B8%8A%202495.txt"
+    ]
+
+
+class ThaiConstitutionDataset(datasets.GeneratorBasedBuilder):
+    """Thailand's constitutional archive since 1932"""
+
```
Comment (yongzx) [inline seacrowd/sea_datasets/thai_constitution/thai_constitution.py:69]: define `SEACROWD_SCHEMA_NAME = 'ssp'`

File: seacrowd/sea_datasets/thai_constitution/thai_constitution.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+            version=datasets.Version(_SOURCE_VERSION),
+            description="Thai constitution source schema",
+            schema="source",
+            subset_id="thai_constitution",
+        ),
+        SEACrowdConfig(
+            name="thai_constitution_seacrowd_ssp",
+            version=datasets.Version(_SEACROWD_VERSION),
+            description="Thai constitution SEACrowd schema",
+            schema="seacrowd_ssp",
```
Comment (yongzx) [inline seacrowd/sea_datasets/thai_constitution/thai_constitution.py:82]: replace `ssp` with `self.SEACROWD_SCHEMA_NAME`

File: seacrowd/sea_datasets/thai_constitution/thai_constitution.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "text": datasets.Value("string"),
+                }
+            )
+        elif self.config.schema == "seacrowd_ssp":
```
Comment (yongzx) [inline seacrowd/sea_datasets/thai_constitution/thai_constitution.py:95]: replace `ssp` with `self.SEACROWD_SCHEMA_NAME`

File: seacrowd/sea_datasets/thai_constitution/thai_constitution.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+                    if self.config.schema == "source":
+                        yield (
+                            counter,
+                            {
+                                "id": str(counter),
+                                "text": line.strip(),
+                            },
+                        )
+                    elif self.config.schema == "seacrowd_ssp":
```
Comment (yongzx) [inline seacrowd/sea_datasets/thai_constitution/thai_constitution.py:134]: replace `ssp` with `self.SEACROWD_SCHEMA_NAME`

**Review by holylovenia — APPROVED** [review]
Thanks, LGTM! After @ilhamfp addresses @yongzx's feedback, we can merge this.

**Comment by ilhamfp** [comment]
Updated 👍

**Review by yongzx — APPROVED** [review]

**Pull request merged**